Utilities/Remove-OldAzureVMs.ps1

```diff
@@ -164,16 +165,47 @@ if ($RemoveAutoCleanupVMs) {
 $cleanupRGScriptBlock = {
     $RGName = $args[0]
     $DryRun = $args[1]
+    $RemoveDisks = $args[2]
     if (-not $DryRun) {
+        if ($RemoveDisks) {
+            $vms = Get-AzVM -ResourceGroupName $RGName
+        }
         Remove-AzResourceGroup -Name $RGName -Verbose -Force
+
+        if ($? -and $RemoveDisks) {
+            foreach ($vm in $vms) {
+                $osDiskUri = $vm.StorageProfile.OSDisk.Vhd.Uri
+                if ($osDiskUri) {
+                    $osDiskContainerName = $osDiskUri.Split('/')[-2]
+                    $osDiskStorageAcct = Get-AzStorageAccount | where { $_.StorageAccountName -eq $osDiskUri.Split('/')[2].Split('.')[0] }
+                    Write-Host "Remove unmanaged OS disk $osDiskUri"
+                    $osDiskStorageAcct | Remove-AzStorageBlob -Container $osDiskContainerName -Blob $osDiskUri.Split('/')[-1]
+                } else {
+                    Write-Host "Remove managed OS disk $($vm.StorageProfile.OsDisk.Name)"
+                    Get-AzDisk -DiskName $vm.StorageProfile.OsDisk.Name  | Remove-AzDisk -Force
```

Comment:
Managed disks are automatically deleted when removing the resource group. You don't need to delete here.
Please try and confirm this.

Reply:
Yes, but in case, it refer to disks under different RG, and I tested, even it already is deleted, this line won't throw error, just do nothing.